src/main/java/com/google/sps/servlets/schedule/ScheduleHandlerServlet.java

```diff
@@ -152,12 +123,154 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
       List<String> resources = new ArrayList<String>(Arrays.asList(links.split(",")));
       
-      UserPreferences.resources = resources;
+      // Get access token
+      String accessToken = (String) request.getSession(false).getAttribute("access_token");
+      // Get primary ID
+      GetCalendar getCalendar = new GetCalendar();
+      String json = new HTTP().get(getCalendar.createGetCalendarURL("primary", accessToken));
+      JSONObject jsonObject = new HTTP().parseJSON(json);
+      String primary_id = (String) jsonObject.get("id");
+
+      if (DB.getUser(primary_id) == null) {
+        response.sendRedirect("/request-permission");
+        return;
+      }
+
+      Gson g = new Gson();
+      json = g.toJson(resources);
+      DB.updateTextProperty(primary_id, "resources", json);
+      
 
       response.getWriter().print(true);
 
     } catch (Exception e) {
-      System.out.println("There was an error trying to get the resources!" + e);
+      System.out.println("There was an error trying to get the resources! " + e);
+      try {
+        response.sendRedirect("/home.html");
+        return;
+      } catch (Exception err) {
+        System.out.println(err);
+      }
+    }
+  }
+
+  /** getInputAndSaveToDB: Saves input to DB. 
+   * 
+   * @param request 
+   * @param response go back to home.html
+   * @return none simply fill to database
+  */
+  public void getInputAndSaveToDB(HttpServletRequest request, HttpServletResponse response) {
+    try {
+      // Get access token
+      String accessToken = (String) request.getSession(false).getAttribute("access_token");
+
+      // Get primary ID
+      GetCalendar getCalendar = new GetCalendar();
+      String json = new HTTP().get(getCalendar.createGetCalendarURL("primary", accessToken));
+      JSONObject jsonObject = new HTTP().parseJSON(json);
+      String primary_id = (String) jsonObject.get("id");
+
+      
+      // UPDATE Start Day
+      int startDay = Integer.parseInt(request.getParameter("startDay"));
+      Gson g = new Gson();
+      json = g.toJson(startDay);
+      DB.updateTextProperty(primary_id, "startDay", json);
+
+
+      // UPDATE Start Week
+      int startWeek = Integer.parseInt(request.getParameter("startWeek"));
+      json = g.toJson(startWeek);
+      DB.updateTextProperty(primary_id, "startWeek", json);
+
+      // UPDATE Difficulty of schedule
+      int intensity = Integer.parseInt(request.getParameter("intensity"));
+      json = g.toJson(intensity);
+      DB.updateTextProperty(primary_id, "userEventsChoice", json);
+
+      // UPDATE Description
+      String user_description = request.getParameter("description");
+      json = g.toJson(user_description);
+      DB.updateTextProperty(primary_id, "description", user_description);
+
+      // UPDATE possible study session days
+      int studySessionDays = Integer.parseInt(request.getParameter("days"));
+      List<Integer> days = new ArrayList<Integer>();
+      days.add(1);
```

Comment:
Similar to my comment on a previous CL, use an array to iterate through these values next time to reduce the long blocks of similar code.

Reply:
Right!